docs/how-to-use-hipblaslt-offline-tuning.rst

```diff
@@ -0,0 +1,85 @@
+
+It has a command line interface. (See :ref:`clients` for more details.)
+
+Through the User Offline Tuning mechanism, the best solution index for a given problem size obtained by ``hipblaslt-bench`` can be directly used in future GEMM calls.
+
+However, please note that these indices cannot be reused across library releases or across different device architectures.
+
+Running
+=================================
+
+Follow these steps:
+
+1. Generate tuning command line by setting environment variable ``HIPBLASLT_LOG_MASK=32`` before calling hipBLASLt APIs if you are unsure how to use ``hipblaslt-bench``. (see :ref:`Logging and heuristics <logging-heuristics>` for more details.)
+
+- In Bash shell:
+
+.. code-block:: bash
+
+    export HIPBLASLT_LOG_MASK=32
+
+- Using `sample_hipblaslt_gemm.cpp <https://github.com/ROCm/hipBLASLt/blob/develop/clients/samples/01_basic_gemm/sample_hipblaslt_gemm.cpp>`_  as an example:
+
+.. code-block:: bash
+
+    ./sample_hipblaslt_gemm
+
+- The tuning command will log:
+
+.. code-block:: bash
+
+    hipblaslt-bench --api_method c -m 1024 -n 512 -k 1024 --lda 1024 --ldb 1024 --ldc 1024 --ldd 1024  --stride_a 0 --stride_b 0 --stride_c 0 --stride_d 0  --alpha 1.000000 --beta 1.000000 --transA N --transB N --batch_count 1  --a_type f16_r --b_type f16_r --c_type f16_r --d_type f16_r --scale_type f32_r --bias_type f32_r   --compute_type f32_r 
+
+
+2. Set the environment variable ``HIPBLASLT_TUNING_FILE=<file_name>`` to tune and store the tuning result of the best solution indices for those GEMM problems, where ``<file_name>`` points to the tuning file.
+
+- Please note that the default settings will have the following changes in the tuning environment.
+
+.. code-block:: bash
+
+    --iters |-i <value>             Iterations to run inside timing loop                                                (Default value is: 1000)
+    --cold_iters |-j <value>        Cold Iterations to run before entering the timing loop                              (Default value is: 1000)
+    --algo_method <value>           Use different algorithm search API. Options: heuristic, all, index (still considered all).                  (Default value is: all)
+    --requested_solution <value>    Requested solution num. Set to -1 to get all solutions. Only valid when algo_method is set to heuristic.    (Default value is: -1)
+    --rotating <value>              Use rotating memory blocks for each iteration, size in MB.                          (Default value is: 512)
+
+- After the tuning completes, the expected output is as follows:
+
+.. code-block:: bash
+
+    ./hipblaslt-bench --api_method c -m 64 -n 64 -k 1024 --lda 64 --ldb 1024 --ldc 64 --ldd 64  --stride_a 0 --stride_b 0 --stride_c 0 --stride_d 0  --alpha 1.100000 --beta 0.900000 --transA N --transB N --batch_count 1  --a_type f32_r --b_type f32_r --c_type f32_r --d_type f32_r --scale_type f32_r --bias_type f32_r   --compute_type f32_r
```

Comment:
with yangwen's PR, the actual output of MASK=32 would contain the solution_index.